src/chem/viirs_aod2ioda.py

```diff
@@ -10,42 +11,74 @@
 import argparse
 import netCDF4 as nc
 import numpy as np
-from datetime import datetime, timedelta
+from datetime import datetime, date, timedelta
+import os
 from pathlib import Path
 
 IODA_CONV_PATH = Path(__file__).parent/"@SCRIPT_LIB_PATH@"
 if not IODA_CONV_PATH.is_dir():
     IODA_CONV_PATH = Path(__file__).parent/'..'/'lib-python'
 sys.path.append(str(IODA_CONV_PATH.resolve()))
 
-import ioda_conv_ncio as iconv
+import ioda_conv_engines as iconv
+from collections import defaultdict, OrderedDict
 from orddicts import DefaultOrderedDict
 
-vName = {
-    'A': "aerosol_optical_depth_4",
-}
-
 locationKeyList = [
     ("latitude", "float"),
     ("longitude", "float"),
     ("datetime", "string")
 ]
 
-AttrData = {}
 
+obsvars = {
+    'A': "aerosol_optical_depth_4",
+}
 
-class AOD(object):
+AttrData = {
+    'converter': os.path.basename(__file__),
+    'nvars': np.int32(len(obsvars)),
+}
 
-    def __init__(self, filename, method, mask, thin, writer):
+
+DimDict = {
+}
+
+VarDims = {
+    'aerosol_optical_depth_4': ['nlocs']
+}
+
+
+class AOD(object):
+    def __init__(self, filename, method, mask, thin):
+        print('filename= ', filename)
+        print('pwd=', os.getcwd())
         self.filename = filename
-        self.method = method
+        print('self.filename= ', self.filename)
         self.mask = mask
+        self.method = method
         self.thin = thin
-        self.data = DefaultOrderedDict(lambda: DefaultOrderedDict(dict))
-        self.writer = writer
+        self.varDict = defaultdict(lambda: defaultdict(dict))
+        self.outdata = defaultdict(lambda: DefaultOrderedDict(OrderedDict))
+        self.varAttrs = DefaultOrderedDict(lambda: DefaultOrderedDict(dict))
         self._read()
 
     def _read(self):
+        # set up variable names for IODA
+        for ncvar, iodavar in obsvars.items():
+            self.varDict[iodavar]['valKey'] = iodavar, iconv.OvalName()
+            self.varDict[iodavar]['errKey'] = iodavar, iconv.OerrName()
+            self.varDict[iodavar]['qcKey'] = iodavar, iconv.OqcName()
+
+            self.varAttrs[iodavar, iconv.OvalName()]['coordinates'] = 'longitude latitude'
+            self.varAttrs[iodavar, iconv.OerrName()]['coordinates'] = 'longitude latitude'
+            self.varAttrs[iodavar, iconv.OqcName()]['coordinates'] = 'longitude latitude'
```

Comment:
If the data provider has defined missing values for any of these variables, please specify those using the _FillValue variable attribute. Use "_FillValue" for the attribute name, and use the same data type as the variable for the attribute data type. Thanks!

Reply:
I'm assuming that there are no fill values defined by the data provider, which is fine.